File: Kontes pribadi/keuntungandagang.pas
var
	A,jumlah: array[0..100] of Integer;
	n,m,k,i,penghasilan: Integer;

begin
	readln(n,m);
	jumlah[0]:=0;
	for i:=1 to n do 
		begin
		if (i<>n) then
			read(A[i])
		else
			readln(A[i]);
		jumlah[i]:=jumlah[i-1]+A[i];
		end;
	readln(k);
	if (k<n) then
		begin
			penghasilan:=jumlah[k];
			writeln(penghasilan);
		end
	else if (k=n) then
		begin
			penghasilan:=jumlah[k];
			writeln(penghasilan);
		end
	else if (k>n) then
		begin
			penghasilan:=jumlah[n]-(m*(k-n));
			if (penghasilan<0) then
				writeln('Pak Dengklek Rapopo')
			else
				writeln(penghasilan);
		end;

end.

File: 02 - SMA/Kontes pribadi/keuntungandagang.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
	int n,m,k,i,penghasilan;
	int A[101],jumlah[101];
	
	jumlah[0]=0;
	cin>>n>>m;
	for (i=1;i<=n;i++){
		scanf("%d",&A[i]);
		jumlah[i]=jumlah[i-1]+A[i];
	}
	scanf("%d",k);
	
	if (k<n){
		penghasilan=jumlah[k];
		printf("%d\n",penghasilan);
		}
	else if (k=n){
		penghasilan=jumlah[n];
		printf("%d\n",penghasilan);
		}
	else if (k>n){
		penghasilan=(jumlah[n]-(m*(k-n)));
			if (penghasilan<0)
				cout<<"Pak Dengklek Rapopo"<<endl;
			else if (penghasilan>=0)
				cout<<penghasilan<<endl;
	}
	printf("%d\n",k);
	
	return 0;
}
